src/core/src/pass/manager.cpp

```diff
@@ -89,89 +114,174 @@ class stopwatch {
+            stopwatches[name].start();
 
-        if (auto matcher_pass = dynamic_pointer_cast<MatcherPass>(pass)) {
-            // This checks is to skip the graph transformation when the graph pass relies on
-            // static shape but the function state is dynamic.
-            if (matcher_pass->get_property(PassProperty::REQUIRE_STATIC_SHAPE) && func->is_dynamic()) {
-                OPENVINO_DEBUG("Pass ",
-                               pass->get_name(),
-                               " requires static shape but the "
-                               "model is dynamic. Skipping this transformation");
-                continue;
-            }
-            // GraphRewrite is a temporary container for MatcherPass to make execution
-            // on on entire ov::Model
-            pass_applied = GraphRewrite(matcher_pass).run_on_model(func);
-        } else if (auto function_pass = dynamic_pointer_cast<ModelPass>(pass)) {
-            // This checks is to skip the graph transformation when the graph pass relies on
-            // static shape but the function state is dynamic.
-            if (function_pass->get_property(PassProperty::REQUIRE_STATIC_SHAPE) && func->is_dynamic()) {
-                OPENVINO_DEBUG("Pass ",
-                               pass->get_name(),
-                               " requires static shape but the "
-                               "model is dynamic. Skipping this transformation");
-                continue;
+            bool is_pass_manager = name == m_manager_name;
+            if (is_pass_manager) {
+                std::cout << "PassManager " << m_manager_name << " started: " << std::endl;
             }
+        }
+    }
+
+    void stop_timer(const std::string& name, bool applied) {
+        if (m_profile_pass.is_enabled()) {
+            auto& stopwatch = stopwatches.at(name);
+            stopwatch.stop();
 
-            if (dynamic_pointer_cast<Validate>(pass)) {
-                if (needs_validate) {
-                    function_pass->run_on_model(func);
-                    needs_validate = false;
+            bool is_pass_manager = name == m_manager_name;
+            if (m_profile_pass.is_bool()) {
+                if (is_pass_manager) {
+                    std::cout << "PassManager: ";
+                } else {
+                    std::cout << "Transformation: ";
+                }
+                std::cout << name << " " << stopwatch.get_milliseconds() << "ms " << (applied ? "+" : "-") << std::endl;
+            } else if (m_file.is_open()) {
+                if (is_pass_manager) {
+                    m_file << "m;" << name << ";" << stopwatch.get_timer_value().count() << ";"
+                           << (applied ? "1;" : "0;") << std::endl;
+                } else {
+                    m_file << "t;" << name << ";" << m_manager_name << ";" << stopwatch.get_timer_value().count() << ";"
+                           << (applied ? "1" : "0") << ";" << std::endl;
```

Comment:
python3 csv parser recognizes the last field - field after the last ';'. So currently the last field will be empty. It is not needed.